src/_pytest/main.py

```diff
@@ -478,6 +479,9 @@ def __init__(self, config: Config) -> None:
         self.trace = config.trace.root.get("collection")
         self._initialpaths: FrozenSet[Path] = frozenset()
 
+        self._pm = config.pluginmanager
+        self._num_conftest_plugins_collected = 0
+        self._fspath_hookproxy_cache: Dict[os.PathLike[str], PathAwareHookProxy] = {}
```

Comment:
Can we add a quick comment here explaining why the cache exists? Something like:

```python
# Cache of hookproxy objects, which speeds up collection. The cache is cleared whenever
# the number of collected conftest changes.
# See #9125 for a detailed summary of the gains.
```

Reply:
i think we need to use a union of HookCaller and PathAwareHookProxy here and in the function to spell the type correct

@bluetech any idea why typing hasn't picked this up

Reply:
I am getting different results when trying to import `HookCaller` It seems that `HookCaller` is defined inside [`pluggy.hooks._HookCaller` in v0.13.1](https://github.com/pytest-dev/pluggy/blob/b1b07a309653a42a5311810f71d0addd09a4ec81/src/pluggy/hooks.py#L201) and as [`pluggy._hooks._HookCaller` in v1.0.0](https://github.com/pytest-dev/pluggy/blob/4259fdd799d64f953631e47ddcb69c3074a15c13/src/pluggy/_hooks.py#L184)

`setup.cfg` defines pluggy dependency `pluggy>=0.12,<2.0`  in such case this dependency is no longer valid right ? or maybe i am importing the wrong `_HookCaller`